tao_compiler/mlir/disc/transforms/fusion_utils.h

```diff
@@ -733,21 +782,34 @@ class StitchGpuFusionStrategy : public FusionStrategy {
       Operation* op, SmallVector<std::pair<Value, TileInfo>, 4>& in_info,
       bool& cover);
   bool findFusionPatternTypeAndSubroot(ShapeAnalysis& shapeAnalysis,
-                                       FusionPattern& fused_pattern);
+                                       FusionPattern& fusion_pattern);
   bool tileXroots(ShapeAnalysis& shapeAnalysis, FusionPattern& fusion_pattern);
   bool backtraceTileAndCover(ShapeAnalysis& shapeAnalysis,
                              FusionPattern& fusion_pattern, Value value);
 };
 
+class PreDotGpuFusionStrategy : public BaseFusionStrategy {
```

Comment:
why this strategy is introduced?

Reply:
So, it means do fusion before DotFusion, not fusing ops before DotOp ... 😂
It look OK but why not make DotFusion more powerful to match multiple element-wise op after DotOp ?